test/webserver.mjs

```diff
@@ -188,154 +194,148 @@ WebServer.prototype = {
       if (verbose) {
         console.log(url);
       }
-      serveRequestedFile(filePath);
+      self.#serveFile(res, filePath, fileSize);
     }
+  }
 
-    function escapeHTML(untrusted) {
+  #serveDirectoryIndex(response, pathPart, queryPart, directory) {
+    const escapeHTML = untrusted =>
       // Escape untrusted input so that it can safely be used in a HTML response
       // in HTML and in HTML attributes.
-      return untrusted
+      untrusted
         .replaceAll("&", "&amp;")
         .replaceAll("<", "&lt;")
         .replaceAll(">", "&gt;")
         .replaceAll('"', "&quot;")
         .replaceAll("'", "&#39;");
-    }
 
-    function serveDirectoryIndex(dir) {
-      res.setHeader("Content-Type", "text/html");
-      res.writeHead(200);
+    response.setHeader("Content-Type", "text/html");
+    response.writeHead(200);
 
-      if (queryPart === "frame") {
-        res.end(
-          "<html><frameset cols=*,200><frame name=pdf>" +
-            '<frame src="' +
-            encodeURI(pathPart) +
-            '?side"></frameset></html>',
-          "utf8"
-        );
+    if (queryPart === "frame") {
+      response.end(
+        `<html>
+          <frameset cols=*,200>
+            <frame name=pdf>
+            <frame src="${encodeURI(pathPart)}?side">
+          </frameset>
+        </html>`,
+        "utf8"
+      );
+      return;
+    }
+
+    const all = queryPart === "all";
+    fs.readdir(directory, function (error, files) {
+      if (error) {
+        response.end();
```

Comment:
Given that we require at least Node.js version 18, would it be feasible to make `#serveDirectoryIndex` async such that we can use the [promise-version](https://nodejs.org/api/fs.html#fspromisesreaddirpath-options) of the `fs` module instead?
That might lead to nicer code here, since we could then do something like below, which avoids the callback-structure and reduces indentation.
```suggestion
    let files;
    try {
      files = await fsPromises.readdir(directory);
    } catch {
      response.end();
      return;
    }
    response.write(
      ...
```

Reply:
Good idea; done in the new commit.